jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java

```diff
@@ -47,74 +50,118 @@ public Asn1DecodeMessageJSON(OdeProperties odeProps) {
 	protected Object process(String consumedData) {
 		OdeData odeData = null;
 		OdeMsgPayload payload = null;
-		OdeBsmMetadata metadata = null;
 		try {
 			JSONObject rawJSONObject = new JSONObject(consumedData);
 			Set<?> keys = rawJSONObject.keySet();
 			for (Object key : keys) {
 				
 				//Send encoded BSM content to Codec service to decode BSM
```

Comment:
There is lot of code duplication below, there is an opportunity for some refactoring.  OK, as if for now.

Reply:
Agreed, and work with Drew on the TIMs (so now we have 3 messages, with many more to come) to plan the standard approach.